imports/collections/users_private.js

```diff
@@ -18,6 +18,29 @@ Meteor.methods({
         console.log(user);
     },
 
+    'users.editProfile'(userId, eMail, longName) {
+        if (Meteor.isServer) {
+            check(eMail, String);
+            check(longName, String);
+            if (! Meteor.userId()) {
+                throw new Meteor.Error("Cannot edit profile", "User not logged in.");
+            }
+
+            if (!Meteor.user().isAdmin) {
+                if (Meteor.userId() != userId) {
+                    throw new Meteor.Error("Cannot edit profile", "You are not admin or you are trying to change someone else's profile");
+                }
+            }
+            
+            if (! emailAddressRegExpTest.test(eMail)) {
+                throw new Meteor.Error("Invalid E-Mail", "Not a valid E-Mail address");
+            }
+            
+            Meteor.users.update(userId, {$set: {'emails.0.address': eMail, 'profile.name': longName}});
```

Comment:
I would suggest to add a check for `Meteor.user().isLDAPuser` - and throw an exception, if true. Because with the current implementation an LDAP user can not use the Menu (good!), but he can call the Meteor method directly from his browser console (bad!). An LDAP user should not be able to change the long name or his mail address.